web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -493,14 +667,333 @@ $.fn.drawer = function(id, ajax) {
+    }
+
+    /* For some reason on IOS5 if you use the jQuery show method it
+     * doesn't display the JS validation error messages unless you do this
+     * or you cause a screen redraw by changing the phone orientation.
+     * NB: This has to happen after the call to show() in fixmystreet.update_pin */
+    if ( navigator.userAgent.match(/like Mac OS X/i)) {
+        document.getElementById('side-form').style.display = 'block';
     }
+    $('#side').hide();
 
+    if (fixmystreet.map.updateSize) {
+        fixmystreet.map.updateSize(); // required after changing the size of the map element
+    }
+    if (fixmystreet.map.panTo) {
+        fixmystreet.map.panDuration = 100;
+        fixmystreet.map.panTo(lonlat);
+        fixmystreet.map.panDuration = 50;
+    }
+
+    $('#sub_map_links').hide();
+    if ($('html').hasClass('mobile')) {
+        var $map_box = $('#map_box'),
+            width = $map_box.width(),
+            height = $map_box.height();
+        $map_box.append(
+            '<p id="mob_sub_map_links">' +
+            '<a href="#" id="try_again">' +
+                translation_strings.try_again +
+            '</a>' +
+            '<a href="#ok" id="mob_ok">' +
+                translation_strings.ok +
+            '</a>' +
+            '</p>')
+        .css({
+            position: 'relative', // Stop map being absolute, so reporting form doesn't get hidden
+            width: width,
+            height: height
+        });
+
+        $('.mobile-map-banner').html('<a href="/">' + translation_strings.home + '</a> ' + translation_strings.right_place);
+
+        // mobile user clicks 'ok' on map
+        $('#mob_ok').toggle(function(){
+            //scroll the height of the map box instead of the offset
+            //of the #side-form or whatever as we will probably want
+            //to do this on other pages where #side-form might not be
+            $('html, body').animate({ scrollTop: height-60 }, 1000, function(){
+                $('#mob_sub_map_links').addClass('map_complete');
+                $('#mob_ok').text(translation_strings.map);
+            });
+        }, function(){
+            $('html, body').animate({ scrollTop: 0 }, 1000, function(){
+                $('#mob_sub_map_links').removeClass('map_complete');
+                $('#mob_ok').text(translation_strings.ok);
+            });
+        });
+    }
+
+    fixmystreet.page = 'new';
+  },
+
+  report: function(reportPageUrl, reportId, callback) {
+    $.ajax(reportPageUrl).done(function(html, textStatus, jqXHR) {
+        var $reportPage = $(html);
+        var $sideReport = $reportPage.find('#side-report');
+
+        if ($sideReport.length) {
+            $('#side').hide(); // Hide the list of reports
+            $('#side-report').remove(); // Remove any existing report page content from sidebar
+            $sideReport.appendTo('#map_sidebar'); // Insert this report's content
+
+            var found = html.match(/<title>([\s\S]*?)<\/title>/);
```

Comment:
Any reason for the regex here, rather than pulling out out of the DOM we created further up? Like: `$reportPage.find('head title').text()`? I guess for speed, maybe?